Automattic/pocket-casts-ios
# Integrate onboarding flow when claiming guest pass

| 📘 Part of: #2083  |  
|:---:|

Fixes # 



Adds onboarding flow to the claim guest pass flow when user is not logged in.
| 1 | 2 | 3 | 4 | 
| - | - | - | - |
| ![Simulator Screenshot - iPhone 16 Pro - 2024-09-30 at 14 53 44](https://github.com/user-attachments/assets/54c54149-8d68-45a7-a30b-4ad7d4be68af) | ![Simulator Screenshot - iPhone 16 Pro - 2024-10-01 at 13 33 15](https://github.com/user-attachments/assets/4d5650de-7efb-4fff-b225-3fad5a8842bc) |  ![Simulator Screenshot - iPhone 16 Pro - 2024-10-01 at 13 33 12](https://github.com/user-attachments/assets/ca962356-68fc-43d4-8795-b09c25e5da6b) | ![Simulator Screenshot - iPhone 16 Pro - 2024-10-01 at 13 33 24](https://github.com/user-attachments/assets/dccfd8ea-deef-47c2-a980-f29f97829af2) | 

## To test

- Start the app
- Ensure that referrals FF is enabled
- Using a account that as a Post subscription 
- Go to profile tab
- Tap on the top left gift icon
- Share the guest pass to the `Reminders` app and save the referral URL
- Now logout from your account
- Switch out to the `Reminders` app and tap on the referral URL
- Check if the Claim Guest Pass open in PocketCasts
- Tap on `Activate My Pass`
- Check if the Login screen is shown
- Tap on Not Now
- Check if a Toast is show with a message saying: ` `
- Tap on `Activate My Pass` again
- Login to your account (one without a subscription)
- Check if after login the process continues to claim the guest pass: Should see IAP flow or a message saying that you already redeem
- Now repeat the process but create a new account and check if the claim flow continues to the IAP flow

## Checklist
